src/plone/api/tests/test_content.py

```diff
@@ -912,6 +912,13 @@ def test_find(self):
         documents = api.content.find(portal_type="Document")
         self.assertEqual(len(documents), 2)
 
+    def test_find(self):
```

Comment:
Hi @gogobd, good idea :) 
This test masks the one defined in line 908. It should have another name. In addition, it would be good to verify we find a different number of results in case we ask for an unrestricted search :)

Reply:
The function was renamed, which is good.  Well, I see a typo in the name: `test_unTrestricted_find`.

But it still finds the same number of results with or without `unrestricted=True`, so it does not tell us much.  Can you add a document that is not found in `test_find`, but is found in `test_unrestricted_find`?  I guess set an expiry date.

Reply:
@mauritsvanrees An expiry date in the past did't do it; I don't know how to create an object that is only found with "unrestricted=True"...

Reply:
I tried .start and .end and .effective and .expires; none of those made a difference...

Reply:
@gogobd The catalog's searchResults method does not filter by effective/expires if the current user has the "Access inactive portal content" permission (see https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/CatalogTool.py#L405), so it won't work if your test is running with the Manager role.

I'd suggest:
1. create a document that you do not publish
2. use setRoles to switch current roles to Anonymous
3. now the document should be missing with unrestricted=False, because it's filtered out based on the allowedRolesAndUsers index (https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/CatalogTool.py#L403)

Reply:
I just logged out and that did the trick!